examples/hitl/rearrange_v2/world.py

```diff
@@ -97,6 +124,102 @@ def reset(self) -> None:
                         if link_index:
                             sim_utilities.close_link(ao, link_index)
 
+        # Object id <-> handle mapping
+        self._id_to_handle: Dict[int, str] = {}
+        self._handle_to_id: Dict[str, int] = {}
+        all_objects = sim_utilities.get_all_objects(sim)
+        for obj in all_objects:
+            self._id_to_handle[obj.object_id] = obj.handle
+            self._handle_to_id[obj.handle] = obj.object_id
+
+        # Try initializing object state and metadata information.
+        self._try_initialize_object_states(sim, self._config)
+
+    def update(self, dt: float) -> None:
+        """
+        Update the world state. Call every frame.
+        """
+        sim = self._sim
+        osm = self._object_state_machine
+
+        if osm is not None:
+            osm.update_states(sim, dt)
+            self._state_snapshot_dict = osm.get_snapshot_dict(sim)
+
+    def get_state_snapshot_dict(self) -> Dict[str, Dict[str, Any]]:
+        """
+        Get a snapshot of the current state of all objects.
+
+        Example:
+        {
+            "is_powered_on": {
+                "my_lamp.0001": True,
+                "my_oven": False,
+                ...
+            },
+            "is_clean": {
+                "my_dish.0002:" False,
+                ...
+            },
+            ...
+        }
+        """
+        return self._state_snapshot_dict
+
+    def get_all_states(self) -> Dict[str, ObjectStateSpec]:
+        """
+        Get a map of all object states active in this world.
+        The key is the state name, and the value is the state specification.
+        """
+        return self._all_states
+
+    def get_states_for_object_handle(
+        self, object_handle: str
+    ) -> List[ObjectStateInfo]:
+        """
+        Get a list of all object states for a given object handle.
+        """
+        object_states: Dict[str, Any] = {}
+        state_snapshot = self.get_state_snapshot_dict()
+
+        for state_id, handles in state_snapshot.items():
+            if object_handle in handles:
+                object_states[state_id] = handles[object_handle]
+
+        state_infos: List[ObjectStateInfo] = []
+        all_object_states = self.get_all_states()
+        for state, value in object_states.items():
+            state_spec = all_object_states[state]
```

Comment:
Is this like a semantically understandable string to name the state?

Reply:
This contains [this information](https://github.com/facebookresearch/habitat-lab/blob/main/habitat-lab/habitat/sims/habitat_simulator/object_state_machine.py#L70).